### ISSUE TITLE
Getting "Error: Cannot find module 'next/headers'" when accessing "api/auth/signin"

### ISSUE THREAD

**Issue by edgarlcs** [nextauthjs/next-auth#5743]
### Question 💬

Hello I am trying to implement Facebook Login but I am getting this error:

`error - Error: Cannot find module 'next/headers'
Require stack:
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next-auth/next/index.js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next-auth/index.js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/.next/server/pages/api/auth/[...nextauth].js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/server/next-server.js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/server/next.js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/server/lib/start-server.js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/cli/next-dev.js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/lib/commands.js
- /Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/bin/next
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
    at Function.mod._resolveFilename (/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/build/webpack/require-hook.js:27:32)
    at Function.mod._resolveFilename (/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/build/webpack/require-hook.js:27:32)
    at Function.mod._resolveFilename (/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/build/webpack/require-hook.js:27:32)
    at Function.Module._load (node:internal/modules/cjs/loader:804:27)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next-auth/next/index.js:15:16)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next-auth/next/index.js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next-auth/index.js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/.next/server/pages/api/auth/[...nextauth].js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/server/next-server.js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/server/next.js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/server/lib/start-server.js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/cli/next-dev.js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/lib/commands.js',
    '/Users/edgarlcs/Documents/GitHub/rifadisimo/rifadisimo/node_modules/next/dist/bin/next'
  ]
}`

### How to reproduce ☕️

Here is the code if you want to try by yourself https://github.com/edgarlcs/rifadisimo

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by Supergamer1337** [comment]
I'm getting the exact same error, but while trying to implement the Credentials provider.

**Comment by Supergamer1337** [comment]
I fixed it by going back to V4.15.1

**Comment by edgarlcs** [comment]
> 



> 



> I fixed it by going back to V4.15.1

All right that fixed that issue but now I am getting:
Error: This action with HTTP GET is not supported by NextAuth.js when trying to get in, any idea on that? what is going on haha

**Comment by FilipSzutkowski** [comment]
Getting the same error both when passing providers and when the providers array is empty.

**Comment by FilipSzutkowski** [comment]
@Supergamer1337 @edgarlcs seems like Next-Auth requires Next.js version 13. Updating Next fixed the problem for me on the newest version of Next-Auth. 

Really weird that it didn't say anything about Next version upon 'npm install'.

**Comment by ThangHuuVu** [comment]
For now, `next@12` users should stay at `v4.15.x` until we fix this issue for `v4.16.x`. The root cause is that in `v4.16.x` we use `next/headers` package from `next@13` to support getting the session in React Server Components. More in [this PR](https://github.com/nextauthjs/next-auth/pull/5741)

**Comment by trungthecelestial** [comment]
same problem with me

**Comment by tortechnocom** [comment]
Same problem stuck with "error - Error: Cannot find module 'next/headers'"

**Comment by smolak** [comment]
> For now, `next@12` users should stay at `v4.15.x` until we fix this issue for `v4.16.x` which is intended for `next@13` users. The root cause is that in `v4.16.x` we use `next/headers` package from `next@13` to support getting the session in React Server Components. More in [this PR](https://github.com/nextauthjs/next-auth/pull/5741)

Shouldn't it be a breaking change then, and so v5? Looks that way. The fact it might not work for Next 13 (fair enough, experimental) shouldn't mean it should be the same for Next 12.x.

**Comment by balazsorban44** [comment]
Sorry for the inconvenience, a patch is coming, you can test it out with https://github.com/nextauthjs/next-auth/pull/5745#issuecomment-1304859690 please report back if it worked for you! :pray: I plan to release this very :soon:

**Comment by Getitdan** [comment]
TL;DR
1) Edit `package.json` to:
`"next-auth": "4.15.1"`
2) Make a reminder to review this issue, and revert when solved.

**Comment by balazsorban44** [comment]
Should be fixed in `4.16.3`

**Comment by tzachbon** [comment]
I'm getting this on `4.18.6`

**Comment by dominik-myszkowski** [comment]
4.18.7 and still the same error... What is going on???

**Comment by B3none** [comment]
> 4.18.7 and still the same error... What is going on???

**If you get this whilst NOT on the sign in page**

I got this error on `4.18.10` when trying to use `unstable_getServerSession` without passing the correct arguments. If you pass those it'll work.

**Comment by balazsorban44** [comment]
Please don't comment on already closed issues, they are not monitored. You can add your reproduction at #6559
